core/services/workflows/v2/config.go

```diff
@@ -41,6 +49,12 @@ type EngineLimits struct {
 	MaxConcurrentCapabilityCalls int
 }
 
+type LifecycleHooks struct {
+	OnInitialized       func(err error)
```

Comment:
will it make sense to add an `OnStart` hook?  rate limits fail the start before `OnInitialized` is ever called.  Was somewhat unexpected.

Reply:
I'll move this inside init() for consistency.